core/logger/pyroscope.go

```diff
@@ -0,0 +1,57 @@
+package logger
+
+import (
+	"runtime"
+
+	"github.com/pkg/errors"
+	"github.com/pyroscope-io/client/pyroscope"
+)
+
+// ErrPyroscopeServerAddrMissing error to be triggered if the server address is missing
+var ErrPyroscopeServerAddrMissing = errors.New("pyroscope server address is missing")
+
+// PyroscopeConfig represents the expected configuration for Pyroscope to properly work
+type PyroscopeConfig interface {
+	PyroscopeServerAddress() string
+	PyroscopeAuthToken() string
+
+	AutoPprofBlockProfileRate() int
+	AutoPprofMutexProfileFraction() int
+}
+
+// StartPyroscope starts continuous profiling of the Chainlink Node
+func StartPyroscope(cfg PyroscopeConfig) (*pyroscope.Profiler, error) {
+	if cfg.PyroscopeServerAddress() == "" {
+		return nil, ErrPyroscopeServerAddrMissing
+	}
+
+	runtime.SetBlockProfileRate(cfg.AutoPprofBlockProfileRate())
+	runtime.SetMutexProfileFraction(cfg.AutoPprofMutexProfileFraction())
+
+	return pyroscope.Start(pyroscope.Config{
+		// Maybe configurable to identify the specific NOP - TBD
+		ApplicationName: "chainlink-node",
+
+		ServerAddress: cfg.PyroscopeServerAddress(),
+		AuthToken:     cfg.PyroscopeAuthToken(),
```

Comment:
Can we also set tags from config? this will become more and more important when we try to identify release profiling snapshots from PR integration tests and local testing on sandbox and testnet vs mainnet on prod

Reply:
We should be able to set tags from config.

Reply:
but it would be good to define how to use the tags, cause you could also simply modify the `app name` so it it something like `chainlink.{env}` or something similar, and have that app dedicated for integration tests or sandbox data...

Reply:
I have added an ENV configuration for us to set the `ENVIRONMENT` tag. This would contain `mainnet` by default, and whoever runs the node would change it properly to what they are doing, and it should be reflected in the Pyroscope instance. Whether it is `testnet`, `integration-tests` or anything, this config should cover those values.

Reply:
Thanks Vyzaldy for adding this. We'll probably need to add more tags for diffing perf. Easy tags to add now are version and commit SHA as they are static and can help pinpoint exact version of profiles. really helpful when we want to do perf diffing for releases and separate release profiles from all other noise

Reply:
Can be added in another PR though